src/components/dialog/dialog.component.js

```diff
@@ -180,6 +180,24 @@ class Dialog extends Modal {
     return null;
   }
 
+  /** Clone the children, pass in value of appliedFixedBottom to toggle style if child is form */
+  renderChildren() {
+    const { children } = this.props;
+
+    if (!children) return null;
+
+    if (typeof children !== 'object') return children;
+
+    const childrenArray = Array.isArray(children) ? children : [children];
+    return childrenArray.map((child, index) => {
+      return React.cloneElement(child, {
+        key: String(index),
```

Comment:
It's generally [not recommended](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) to use an array index as a React key.

Reply:
:white_check_mark: